tst/TurnConnectionFunctionalityTest.cpp

```diff
@@ -263,6 +263,92 @@ namespace com { namespace amazonaws { namespace kinesis { namespace video { name
         freeTestTurnConnection(&pTurnConnection);
     }
 
+    struct TurnCustomData {
+        PTurnConnection pTurnConnection;
+        PBYTE pBuf;
+        UINT32 bufLen;
+        PKvsIpAddress pKvsIpAddress;
+    };
+
+    PVOID testTurnSendDataRoutine(PVOID arg) {
+        struct TurnCustomData * pData = (struct TurnCustomData *) arg;
+        EXPECT_EQ(STATUS_SUCCESS, turnConnectionSendData(pData->pTurnConnection, pData->pBuf, pData->bufLen, pData->pKvsIpAddress));
+
+        return 0;
+    }
+
+    TEST_F(TurnConnectionFunctionalityTest, turnConnectionCallMultipleTurnSendDataInThreads)
+    {
+        PTurnConnection pTurnConnection = NULL;
+        BOOL turnReady = FALSE;
+        KvsIpAddress turnPeerAddr;
+        UINT32 bufLen = 5;
+        UINT32 reqCount = 10;
+        PBYTE *buf;
+        struct TurnCustomData *args;
+        TID *threads;
+        UINT32 i, j;
+
+        buf = (PBYTE*) MEMALLOC(SIZEOF(PBYTE) * reqCount);
+        args = (struct TurnCustomData *) MEMALLOC(SIZEOF(struct TurnCustomData) * reqCount);
+        threads = (TID*) MEMALLOC(SIZEOF(TID) * reqCount);
+
+        initializeTestTurnConnection(&pTurnConnection);
+
+        turnPeerAddr.port = (UINT16) getInt16(8080);
+        turnPeerAddr.family = KVS_IP_FAMILY_TYPE_IPV4;
+        turnPeerAddr.isPointToPoint = FALSE;
+        // random peer 10.1.1.1, we are not actually sending anything to it.
+        turnPeerAddr.address[0] = 0x0A;
+        turnPeerAddr.address[1] = 0x01;
+        turnPeerAddr.address[2] = 0x01;
+        turnPeerAddr.address[3] = 0x01;
+
+        EXPECT_EQ(STATUS_SUCCESS, turnConnectionAddPeer(pTurnConnection, &turnPeerAddr));
+        EXPECT_EQ(STATUS_SUCCESS, turnConnectionStart(pTurnConnection));
+
+        // wait until channel is created
+        while(!turnReady) {
+            MUTEX_LOCK(pTurnConnection->lock);
+            if (pTurnConnection->state == TURN_STATE_READY) {
+                turnReady = TRUE;
+            }
+            MUTEX_UNLOCK(pTurnConnection->lock);
+        }
+
+        EXPECT_TRUE(turnReady == TRUE);
+
+        for (i = 0; i < reqCount; i++) {
+            buf[i] = (PBYTE) MEMALLOC(bufLen);
+            for (j = 0; j < bufLen; j++) {
+                buf[i][j] = i;
+            }
+            args[i].pBuf = buf[i];
+            args[i].bufLen = bufLen;
+            args[i].pKvsIpAddress = &turnPeerAddr;
+            args[i].pTurnConnection = pTurnConnection;
+            EXPECT_EQ(STATUS_SUCCESS, THREAD_CREATE(&threads[i], testTurnSendDataRoutine, (PVOID) &args[i]));
```

Comment:
I would use `std::thread`, you can capture multiple things and don't have to worry about putting in a struct. The PeerConnectionFunctionality tests do that.